interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1025 @@
+#     Software without restriction, including without limitation the rights to use, 
+#     copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the 
+#     Software, and to permit persons to whom the Software is furnished to do so, 
+#     subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies 
+#     or substantial portions of the Software.
+
+"""
+yaml2ck.py: Convert Cantera YAML input files or `Solution` objects to 
+            Chemkin-format mechanisms
+
+Usage:
+    yaml2ck [--input=<file_path or str>]
+            [--mechanism-out=<file_path or str>]
+            [--thermo-out=<file_path or str>]
+            [--transport-out=<file_path or str>]
+            [--reorder-reactions=bool]
+            [--overwrite-files=bool]
+            [--cantera-validate]
+
+Example:
+    yaml2ck --input=chem.yaml --output-path=chem.ck --separate-thermo_file=False
+
+The input keyword is required. It specifies the file to be converted into a 
+Chemkin-format mechanism. If it is a string it can either be a path or a filename.
+A filename will assume the path is the working directory.
+
+The mechanism-out keyword is an optional keyword to specify the name of the output
+mechanism or the full path to write to if that is given. If this keyword is not 
+specified it will default to placing a file of the same name as the input file
+with the extension changed to '.ck' in the working directory.
+
+The thermo-out keyword is an optional keyword to specify the name of the output
+thermodynamics data file or the full path to write to if that is given. If this keyword
+is not specified it will default to the thermodynamics inside the mechanism file. 
+
+The transport-out keyword is an optional keyword to specify the name of the output
+transport data file or the full path to write to if that is given. If this keyword is 
+not specified it will default to placing a file of the same name as the input file
+with the extension changed to '.tran' in the working directory.
+
+Reordering the reactions will reorder the species inside of each reaction in descending
+order of molecular weight. This can be useful to create uniform mechanisms that can
+later be searched easily outside of Cantera.
+
+Setting overwrite_files to True will allow yaml2ck to delete a prior version of a
+mechanism and write a file with the same name. If it is False then yaml2ck will throw
+an error.
+
+The cantera-validate keyword is a boolean choice to test that the converted 
+Chemkin-formatted mechanism will load back into Cantera.
+
+Supported Thermodynamics: 
+    NASA7
+
+Supported Reaction models: 
+    Elementary, Three-Body, Falloff, Pressure-Dependent-Arrhenius, Chebyshev
+
+Unsupported Thermodynamic Property Models:
+    NASA9, Shomate, constant-Cp
+
+Unsupported Reaction models (not comprehensive): 
+    Chemically-activated, Blowers-Masel, surface-Blowers-Masel
```

Comment:
`surface-Blowers-Masel` can be removed, as it will not make it into a stable Cantera version (see #1181). You may also want to clarify whether surface mechanisms are currently supported or not.